wgpu-types/src/lib.rs

```diff
@@ -7423,6 +7437,101 @@ impl Default for InstanceDescriptor {
     }
 }
 
+#[derive(Clone, Debug, PartialEq, Eq)]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+/// Descriptor for all size definiing attributes of a single triangle geometry inside a bottom level acceleration structure.
+pub struct BlasTriangleGeometrySizeDescriptor {
+    /// Format of a vertex position, must be [VertexFormat::Float32x3]
+    /// with just [Features::RAY_TRACING_ACCELERATION_STRUCTURE]
+    /// but later features may add more formats.
+    pub vertex_format: VertexFormat,
+    /// Number of vertices.
+    pub vertex_count: u32,
+    /// Format of an index. Only needed if an index buffer is used.
+    /// If `index_format` is provided `index_count` is required.
+    pub index_format: Option<IndexFormat>,
+    /// Number of indices. Only needed if an index buffer is used.
+    /// If `index_count` is provided `index_format` is required.
+    pub index_count: Option<u32>,
+    /// Flags for the geometry.
+    pub flags: AccelerationStructureGeometryFlags,
+}
+
+#[derive(Clone, Debug)]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+/// Descriptor for all size definiing attributes of all geometries inside a bottom level acceleration structure.
+pub enum BlasGeometrySizeDescriptors {
+    /// Triangle geometry version.
+    Triangles {
+        /// Descriptor for each triangle geometry.
+        desc: Vec<BlasTriangleGeometrySizeDescriptor>,
+    },
+}
```

Comment:
This descriptor should take a slice instead of a vector.

Also, wgpu so far has avoided chopping off words in the public API. I think that it's a good thing, and the plural form is informative.

```suggestion
pub enum BlasGeometrySizeDescriptors<'a> {
    /// Triangle geometry version.
    Triangles {
        /// Descriptor for each triangle geometry.
        descriptors: &'[BlasTriangleGeometrySizeDescriptor],
    },
}
```

For consistency with the rest of the API, BlasGeometrySizeDescriptors should be passed by reference (for example in `create_blas`.

Reply:
This is required to have no lifetimes because it is used in the wgpu-core BLAS here
https://github.com/Vecvec/wgpu/blob/0fcc749efda0f22ccbcd5dffed3442d68fd17850/wgpu-core/src/resource.rs#L1924-L1925
doing this would require a duplicate size descriptor for wgpu-core that was a vector.

Reply:
I suspected something like that. My inclination is that wgpu's `create` functions are usually cold code and that we can pay for an extra allocation and copy for the sake of consistency, after all the internal detail that motivate passing a vector may change in the future. But that's something we can easily revisit later so it's fine to move on and land a first implementation this way.

Reply:
So (just checking) does that mean that this should or shouldn't have something done about it here?

Reply:
Sorry, that was ambiguous. Let's leave it the way it is for now.